src/Validation/ScannableURLProvider.php

```diff
@@ -307,7 +310,7 @@ private function get_posts_by_type( $post_type, $offset = null ) {
 		}
 		$query = new WP_Query( $args );
 
-		return $this->get_posts_that_support_amp( $query->posts );
+		return $this->get_posts_that_support_amp( $query->posts, $posts_to_exclude );
```

Comment:
Instead of updating the `get_posts_that_support_amp` method with this new parameter, it would seem better to me to just do the filtering here in this method:

```suggestion
		$posts = array_filter( $query->posts, static function ( $post ) use ( $posts_to_exclude ) { return ! in_array( $post->ID, $posts_to_exclude, true ); } );
		return $this->get_posts_that_support_amp( $query->posts, $posts_to_exclude );
```

(Apologies for lack of formatting.)

Reply:
Hold on, this can be simplified further since we're getting IDs:

```suggestion
		return $this->get_posts_that_support_amp( array_diff( $query->posts, $posts_to_exclude ) );
```

Reply:
I thought since we already filter these post IDs and `array_diff` is considered slow, completing the task in a single run would be performant. However, any performance impact should be negligible here as the array size will be 100 only.

Reply:
So we can go with `array_diff()` instead? It's less code and doens't change the resulting API.